ipatests/pytest_ipa/integration/create_quarkus.py

```diff
@@ -0,0 +1,153 @@
+                                  "openssl", "maven", "wget",
+                                  "firefox", "xorg-x11-server-Xvfb"])
+    #  add keycloak system user/group and folder
+    url = "https://github.com/keycloak/keycloak/releases/download/{ver}/keycloak-{zip}.zip".format(ver=version, zip=version)  # noqa: E501
+    host.run_command(["wget", url, "-O", "{0}-{1}.zip".format(dir, version)])
+    host.run_command(
+        ["unzip", "{0}-{1}.zip".format(dir, version), "-d", "/opt/"]
+    )
+    host.run_command(["mv", "{0}-{1}".format(dir, version), dir])
+    host.run_command(["groupadd", "keycloak"])
+    host.run_command(["useradd", "-r", "-g", "keycloak", "-d", dir, "keycloak"]
+                     )
+    host.run_command(["chown", "-R", "keycloak:", dir])
+    host.run_command(["chmod", "o+x", "{0}/bin/".format(dir)])
+    host.run_command(["restorecon", "-R", dir])
+
+    # setup TLS certificate using IPA CA
+    host.run_command(["kinit", "-k"])
+    host.run_command(["ipa", "service-add", "HTTP/{0}".format(host.hostname)])
+    key = "/etc/pki/tls/private/keycloak.key"
+    crt = "/etc/pki/tls/certs/keycloak.crt"
+    keystore = "/etc/pki/tls/private/keycloak.store"
+    host.run_command(["ipa-getcert", "request", "-K",
+                      "HTTP/{0}".format(host.hostname),
+                      "-D", host.hostname, "-o", "keycloak",
+                      "-O", "keycloak", "-m", "0600",
+                      "-M", "0644",
+                      "-k", key, "-f", crt, "-w"])
+    host.run_command(["keytool", "-import", "-keystore", keystore,
+                      "-file", "/etc/ipa/ca.crt",
+                      "-alias", "ipa_ca",
+                      "-trustcacerts", "-storepass", password, "-noprompt"])
+    host.run_command(["chown", "keycloak:keycloak", keystore])
+
+    # Setup keycloak service and config files
+    contents = textwrap.dedent("""
+    KEYCLOAK_ADMIN=admin
+    KEYCLOAK_ADMIN_PASSWORD={admin_pswd}
+    KC_HOSTNAME={host}:8443
+    KC_HTTPS_CERTIFICATE_FILE={crt}
+    KC_HTTPS_CERTIFICATE_KEY_FILE={key}
+    KC_HTTPS_TRUST_STORE_FILE={store}
+    KC_HTTPS_TRUST_STORE_PASSWORD={store_pswd}
+    KC_HTTP_RELATIVE_PATH=/auth
+    """).format(admin_pswd=password, host=host.hostname, crt=crt, key=key,
+                store=keystore, store_pswd=password)
+    host.put_file_contents("/etc/sysconfig/keycloak", contents)
+
+    contents = textwrap.dedent("""
+    [Unit]
+    Description=Keycloak Server
+    After=network.target
+
+    [Service]
+    Type=idle
+    EnvironmentFile=/etc/sysconfig/keycloak
+
+    User=keycloak
+    Group=keycloak
+    ExecStart=/opt/keycloak/bin/kc.sh start
+    TimeoutStartSec=600
+    TimeoutStopSec=600
+
+    [Install]
+    WantedBy=multi-user.target
+    """)
+    host.put_file_contents("/etc/systemd/system/keycloak.service", contents)
+    host.run_command(["systemctl", "daemon-reload"])
+    host.run_command(["setenforce", "0"])
```

Comment:
In my tests this was not needed, as long as you relabel content in /opt/keycloak with `restorecon -R /opt/keycloak'.